activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -127,6 +127,10 @@ def initialize(connection, logger = nil, config = {}) # :nodoc:
         )
       end
 
+      def replica?
+        @config[:replica] || false
```

Comment:
@eileencodes Is it in the scope of this PR for each database adapter to verify if the replica database is indeed a replica to help catch user errors? We have this guard in place at Discourse whenever we switch over to a replica database:

```
  value = connection.exec_query("SELECT pg_is_in_recovery()").rows[0][0]
  raise "Replica database server is not in recovery mode." if !value
```

Reply:
At GitHub we have rw databases (primary) and ro databases (replica). The replicas have different users than the primary ones. In addition we also have readonly connections - a rw database can be connected to in a ro way. This change is for the first part - to be able to identify the replica database in the configurations list. Later when I implement the connection switching I'll handle your case as well since we need that at GitHub, but that's going to be a separate PR.

Reply:
That is great to hear :) Thank you for sharing.